sound/soc/sof/intel/hda-ipc.c

```diff
@@ -268,6 +268,36 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 	return ret;
 }
 
+bool is_ipc_irq(struct snd_sof_dev *sdev)
+{
+	bool ret = false;
+	u32 irq_status;
+
+	/* The function can be called at irq thread, so use spin_lock_irq */
+	spin_lock_irq(&sdev->hw_lock);
```

Comment:
@bardliao I'm not sure about this. I dont think we need to use spin_lock_irq(). Back when we're still upstreaming the SOF driver, IIRC the comment upstream was that spin_lock_irq() should only be in the hard interrupt handler.

Reply:
Thanks. I will double check it.

Reply:
@ranj063 I can see spin_lock() is called twice without spin_unlock() if I test it with spin_lock() instead of spin_lock_irq(). I think the reason is that the irq did not be disabled when we use spin_lock() and the irq handler could be triggered when the irq thread is still running. Also, [spin_lock_irq()](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda-stream.c#L618) is using in hda_dsp_stream_threaded_handler() which is a irq thread actually. So I believe that we should use spin_lock_irq()